pennylane/templates/subroutines/qsvt.py

```diff
@@ -28,8 +28,70 @@
 from pennylane.wires import Wires
 
 
-# pylint: disable=too-many-branches, unused-argument
-def qsvt(A, poly, encoding_wires=None, block_encoding=None, **kwargs):
+def _pauli_rep_process(A, poly, encoding_wires, block_encoding):
+    if block_encoding not in ["prepselprep", "qubitization", None]:
+        raise ValueError(
+            "block_encoding = {block_encoding} not supported for A of type {type(A)}. When A is a Hamiltonian or has a Pauli decomposition, block_encoding should take the value 'prepselprep' or 'qubitization'. Otherwise, please provide the matrix of the Hamiltonian as input. For more details, see the 'qml.matrix' function."
+        )
+
+    if any(wire in qml.wires.Wires(encoding_wires) for wire in A.wires):
+        raise ValueError(
+            f"Control wires in '{block_encoding}' should be different from the hamiltonian wires"
+        )
+
+    encoding = (
+        qml.Qubitization(A, control=encoding_wires)
+        if block_encoding == "qubitization"
+        else qml.PrepSelPrep(A, control=encoding_wires)
+    )
+
+    angles = qml.poly_to_angles(poly, "QSVT")
+
+    projectors = [
+        qml.PCPhase(angle, dim=2 ** len(A.wires), wires=encoding_wires + A.wires)
+        for angle in angles
+    ]
+    return encoding, projectors
+
+
+def _tensorlike_process(A, poly, encoding_wires, block_encoding):
+    if block_encoding not in ["embedding", "fable", None]:
+        raise ValueError(
+            "block_encoding = {block_encoding} not supported for A of type {type(A)}. When A is a matrix block_encoding should take the value 'embedding' or 'fable'. Otherwise, please provide an input with a Pauli decomposition. For more details, see the 'qml.pauli_decompose' function."
+        )
```

Comment:
Could you split this string over multiple lines so that it's less than 100 characters per line?

Also, I'm pretty sure this is supposed to be an f-string. Not sure why it wasn't one before 🤔